tls/cfsslscan_handshake.go

```diff
@@ -22,9 +71,51 @@ func (c *Conn) SayHello() (cipherID, version uint16, err error) {
 	if err != nil {
 		return
 	}
+ 	if IsECcipherSuite(serverHello.cipherSuite) {
+		// Prime the connection, if necessary, for key
+		// exchange messages by reading off the certificate
+		// message and, if necessary, the OCSP stapling
+		// message
+		var msg interface{}
+		msg, err = c.readHandshake()
+		if err != nil {
+			return
+		}
+		certMsg, ok := msg.(*certificateMsg)
+		if !ok || len(certMsg.certificates) == 0 {
+			c.sendAlert(alertUnexpectedMessage)
+			err = unexpectedMessageError(certMsg, msg)
+			return
+		}
+		if serverHello.ocspStapling {
```

Comment:
So it doesn't work if you don't read the Certificate message out of the handshake prior to reading ServerKeyExchange, and I feel like the error checks are useful. Given that, I also feel like if there's a stapled CertificateStatus message, it will also need to be read, even if we don't care about it.

I suppose we could simplify the logic in here by removing the type assertions and the checks about them, since they're redundant with logic in readHandshake. But readHandshake still must be called here, and an extra time if the server is doing OCSP Stapling.